common4j/src/test/com/microsoft/identity/common/java/util/CommonURIBuilderTest.java

```diff
@@ -0,0 +1,143 @@
+        Assert.assertEquals("Test1", builder.getQueryParams().get(0).getName());
+        Assert.assertEquals("Value1", builder.getQueryParams().get(0).getValue());
+    }
+
+    @Test
+    public void testCallingAddParametersIfAbsent_WithMap(){
+        final Map<String, String> map = new HashMap<>();
+        map.put("Test1", "Value1");
+        map.put("Test2", "Value2");
+
+        final CommonURIBuilder builder = new CommonURIBuilder().addParametersIfAbsent(map);
+
+        Assert.assertEquals(2, builder.getQueryParams().size());
+        Assert.assertEquals("Test1", builder.getQueryParams().get(0).getName());
+        Assert.assertEquals("Value1", builder.getQueryParams().get(0).getValue());
+        Assert.assertEquals("Test2", builder.getQueryParams().get(1).getName());
+        Assert.assertEquals("Value2", builder.getQueryParams().get(1).getValue());
+    }
+
+    @Test
+    public void testCallingAddParametersIfAbsent_WithMap_ValueAlreadyExists(){
+        final CommonURIBuilder builder = new CommonURIBuilder()
+                .setParameter("Test1", "Value1");
+
+        final Map<String, String> map = new HashMap<>();
+        map.put("Test1", "Value2");
+
+        builder.addParametersIfAbsent(map);
+
+        Assert.assertEquals(1, builder.getQueryParams().size());
+        Assert.assertEquals("Test1", builder.getQueryParams().get(0).getName());
+        Assert.assertEquals("Value1", builder.getQueryParams().get(0).getValue());
+    }
+
+    @Test
+    public void testCallingAddParametersIfAbsent_WithMapEntryList(){
+        final List<Map.Entry<String, String>> map = new ArrayList<>();
+        map.add(new AbstractMap.SimpleEntry<String, String>("Test1", "Value1"));
+        map.add(new AbstractMap.SimpleEntry<String, String>("Test2", "Value2"));
+
+        final CommonURIBuilder builder = new
+                CommonURIBuilder().addParametersIfAbsent(map);
+
+        Assert.assertEquals(2, builder.getQueryParams().size());
+        Assert.assertEquals("Test1", builder.getQueryParams().get(0).getName());
+        Assert.assertEquals("Value1", builder.getQueryParams().get(0).getValue());
+        Assert.assertEquals("Test2", builder.getQueryParams().get(1).getName());
+        Assert.assertEquals("Value2", builder.getQueryParams().get(1).getValue());
+    }
+
+    @Test
+    public void testCallingAddParametersIfAbsent_WithMapEntryList_ValueAlreadyExists(){
+        final CommonURIBuilder builder = new CommonURIBuilder()
+                .setParameter("Test1", "Value1");
+
+        final List<Map.Entry<String, String>> map = new ArrayList<>();
+        map.add(new AbstractMap.SimpleEntry<String, String>("Test1", "Value1"));
+
+        builder.addParametersIfAbsent(map);
+
+        Assert.assertEquals(1, builder.getQueryParams().size());
+        Assert.assertEquals("Test1", builder.getQueryParams().get(0).getName());
+        Assert.assertEquals("Value1", builder.getQueryParams().get(0).getValue());
+    }
+}
```

Comment:
I use `setParameters` because that's what URIBuilder provides. (and I assume that URIBuilder works as expected - the test cases here is for testing our code specifically)

In those tests, we want to verify if adding the same param to an existing object would yield the right result. 
Using `addParametersIfAbsent` as an initial add seems to defeat the purpose.

That said, there are tests, such as `testCallingAddParametersIfAbsent` that tests adding `addParametersIfAbsent` into an empty object - if that's what you're looking for.